CI fails on macOS (late january 2023)

The CI on macOS is currently failing with the following error
```
-- The C compiler identification is AppleClang 14.0.0.14000029
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode_14.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found YARP: /Users/runner/work/bipedal-locomotion-framework/bipedal-locomotion-framework/install/deps/lib/cmake/YARP (found version "3.7.0")
CMake Error at /usr/local/Cellar/cmake/3.25.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Python3_EXECUTABLE Python3_INCLUDE_DIRS
-- Configuring incomplete, errors occurred!
  Python3_LIBRARIES Interpreter Development Development.Module
See also "/Users/runner/work/bipedal-locomotion-framework/bipedal-locomotion-framework/build/CMakeFiles/CMakeOutput.log".
  Development.Embed) (Required is at least version "3.6")
See also "/Users/runner/work/bipedal-locomotion-framework/bipedal-locomotion-framework/build/CMakeFiles/CMakeError.log".

      Reason given by package: 
          Interpreter: Wrong version for the interpreter "/usr/local/bin/python"

Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.25.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPython/Support.cmake:2984 (find_package_handle_standard_args)
  cmake/FindPython3.cmake:[38](https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/4053623191/jobs/6974459708#step:19:39)9 (include)
  cmake/BipedalLocomotionFrameworkFindDependencies.cmake:56 (find_package)
  cmake/BipedalLocomotionFrameworkDependencies.cmake:[52](https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/4053623191/jobs/6974459708#step:19:53) (checkandset_dependency)
  CMakeLists.txt:21 (include)
```

Accordingly to [this](https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/4053623191/jobs/6974459708#step:15:26) the python version installed is 3.11

https://github.com/ami-iit/bipedal-locomotion-framework/actions/runs/4053623191/jobs/6974459708#step:19:68

@traversaro do you have any hints?

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -96,20 +96,21 @@ jobs:
         # Install tomlplusplus
         ./vcpkg.exe --overlay-ports=C:/robotology/robotology-vcpkg-ports install tomlplusplus:x64-windows
 
+    - name: Fixup brew [macOS]
+      if: matrix.os == 'macos-latest'
+      run: |
+          # Unlink and re-link to prevent errors when github mac runner images
+          # install python outside of brew, for example:
+          # https://github.com/orgs/Homebrew/discussions/3895
+          # https://github.com/actions/setup-python/issues/577
+          # https://github.com/actions/runner-images/issues/6459
+          # https://github.com/actions/runner-images/issues/6507
+          # https://github.com/actions/runner-images/issues/2322
+          brew list -1 | grep python | while read formula; do brew unlink $formula; brew link --overwrite $formula; done
+
     - name: Dependencies [macOS]
       if: matrix.os == 'macos-latest'
       run: |
-        # Temporary workaround for https://github.com/actions/setup-python/issues/577
-        rm -f /usr/local/bin/2to3
-        rm -f /usr/local/bin/idle3
-        rm -f /usr/local/bin/pydoc3
-        rm -f /usr/local/bin/python3
-        rm -f /usr/local/bin/python3-config
-        rm -f /usr/local/bin/2to3-3.11
-        rm -f /usr/local/bin/idle3.11
-        rm -f /usr/local/bin/pydoc3.11
-        rm -f /usr/local/bin/python3.11
-        rm -f /usr/local/bin/python3.11-config
         brew update
         brew install ace boost eigen swig qt5 orocos-kdl qhull ipopt cppad pkg-config pybind11 \
                      libmatio spdlog librealsense nlohmann-json numpy assimp

